lib/config/definitions.ts

```diff
@@ -2037,6 +2037,16 @@ const options: RenovateOptions[] = [
     type: 'boolean',
     default: true,
   },
+  {
+    name: 'gitUrl',
+    description:
+      'Overrides the default resolution for git remote, e.g. to switch GitLab from HTTPS to SSH-based. Currently GitLab only. Can be one of `default`, `endpoint` or `ssh`',
+    type: 'string',
+    allowedValues: ['default', 'ssh', 'endpoint'],
+    default: 'default',
+    stage: 'repository',
+    admin: true,
```

Comment:
As far as I can tell, `admin: true` has no effect here. Should that be removed? @viceice

Reply:
FYI this was migrated to `globalOnly: true` now. The point of that flag is also partly for documentation - it indicates to users that it can only be configured in global config and not in-repo config.

Reply:
```suggestion
    globalOnly: true,
```